cluster-provision/okd/scripts/provision.sh

```diff
@@ -90,31 +90,139 @@ if [ ! -f "/etc/installer/token" ]; then
     exit 1
 fi
 
-mkdir -p /root/install
-cp /install-config.yaml /root/install/
+export CLUSTER_DIR=/root/install
+INSTALL_CONFIG_FILE=$CLUSTER_DIR/install-config.yaml
+
+# we need to install jq and yq to make yaml changes
+dnf install -y jq
+pip install yq
+
+mkdir -p $CLUSTER_DIR
+
+# print version
+/openshift-install version
+
+if [[ $INSTALLER_TAG =~ ^.*4\.1$ ]]; then
+    cp /install-config.yaml $CLUSTER_DIR/
+
+elif [[ $INSTALLER_TAG =~ ^.*4\.2$ ]]; then
+    cp /install-config.yaml /tmp/install-config.yaml.tmp
+
+    # set number of workers
+    cat /tmp/install-config.yaml.tmp | yq -y '.compute[].replicas = 2' > "$INSTALL_CONFIG_FILE"
```

Comment:
nitpick: I would use shell redirection `command <filename` instead of `cat filename | command`.